wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 2 - Initial Structure - Split `MySiteViewModel.MySiteUiState` -> `UiModel`

Parent #15215

PS: Don't worry about the size of the PR. It involves only refactoring effort and adds no new functionality.

The PR refactors `MySiteViewModel` to split `MySiteViewModel.MySiteUiState `-> `UiModel` mapping, and moves 
- Cards building logic to new `CardsBuilder` and 
- Dynamic cards building logic to new `DynamicCardsBuilder`.

Site Items are already built using `SiteItemsBuilder` and no changes are made to it.

Tests:  Note that no new tests are added at this point, and only existing card building tests are moved from `MySiteViewModelTest` to respective new builder test classes. Let's continue to add more tests as we find more time.

To test:

- Reviewing by each commit might be the best way to review this PR.
- Smoke test the app to check that the following cards are rendered properly as the refactoring involves the movement of cards building logic to separate builders:
    - `QuickStartCard` or `QuickStartDynamicCard`: Ensure that toggling `quickStartDynamicCardsFeatureConfig`  displays cards appropriately.
    - `SiteInfoCard`
    - `QuickActionsCard` (Visible for WordPress App, hidden for Jetpack app)
    - `DomainRegistrationCard` (Visible for paid plan site when `SiteDomainFeatureConfig` flag enabled, hidden otherwise.) 
        
        ![device-2021-10-08-100224](https://user-images.githubusercontent.com/1405144/136499262-5cc4b88b-0ac6-4f6b-8b3a-b9d0d1a44002.png)


⚠️: The changes are not under any feature flag, and will merge to `develop`. As we're still in the first week of this sprint, we should be able to test it well before they move to `beta`.

## Regression Notes
1. Potential unintended areas of impact 
See above.

2. What I did to test those areas of impact (or what existing automated tests I relied on) 
N/A

3. What automated tests I added (or what prevented me from doing so) 
N/A

PR submission checklist:
